apps/studio/components/interfaces/Realtime/Inspector/RealtimeTokensPopover/index.tsx

```diff
@@ -59,27 +58,30 @@ export const RealtimeTokensPopover = ({ config, onChangeConfig }: RealtimeTokens
   }, [snap.role])
 
   useEffect(() => {
-    let token: string | undefined
-    let bearer: string | null = null
+    const triggerUpdateTokenBearer = async () => {
+      let token: string | undefined
+      let bearer: string | null = null
 
-    if (
-      config.projectRef !== undefined &&
-      jwtSecret !== undefined &&
-      snap.role !== undefined &&
-      snap.role.type === 'postgrest'
-    ) {
-      token = anonRoleKey
-      getRoleImpersonationJWT(config.projectRef, jwtSecret, snap.role)
-        .then((token) => (bearer = token))
-        .catch((err) => toast.error(`Failed to get JWT for role: ${err.message}`))
-    } else {
-      token = serviceRoleKey
+      if (
+        config.projectRef !== undefined &&
+        jwtSecret !== undefined &&
+        snap.role !== undefined &&
+        snap.role.type === 'postgrest'
+      ) {
+        token = anonRoleKey
+        await getRoleImpersonationJWT(config.projectRef, jwtSecret, snap.role)
```

Comment:
Isn't `await` redundant here? Doesn't actually matter, just wondering 😅

Reply:
That's why the role impersonation didn't work, the bearer variable wasn't assigned when calling `onChangeConfig` so it was always null.